### PR TITLE
Refactor selfplay, gtp

### PR DESCRIPTION
**Pull request by brilee**
Some initial renames / cleanup. No significant changes to rl_loop / rl_runner / local_rl_loop yet.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -193,7 +193,7 @@ and you can use any gtp-compliant program with it.
 ```
 # Latest model should look like: /path/to/models/000123-something
 LATEST_MODEL=$(ls -d $MINIGO_MODELS/* | tail -1 | cut -f 1 -d '.')
-BOARD_SIZE=19 python3 main.py gtp -l $LATEST_MODEL --num_readouts=$READOUTS -v 3
+BOARD_SIZE=19 python3 gtp.py --load_file=$LATEST_MODEL --num_readouts=$READOUTS --verbose=3
```
Comment (amj) [inline README.md]: do we need to update the run_local script? And docker image?

File: gtp.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
+# Copyright 2018 Google LLC
```
Comment (amj) [inline gtp.py]: should probably have used `git mv` to preserve history?

File: gtp.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        engine.add_cmd_handler(KgsCmdHandler(player))
+    engine.add_cmd_handler(RegressionsCmdHandler(player))
+    engine.add_cmd_handler(GoGuiCmdHandler(player))
+    engine.add_cmd_handler(MiniguiCmdHandler(player, courtesy_pass=kgs_mode))
+
+    return engine
+
+def main(argv):
+    '''Run Minigo in GTP mode.'''
+    del argv
```
Comment (amj) [inline gtp.py]: worth a comment here about why argv gets parsed twice.

File: README.md (hunk 1 above)
Comment (amj) [reply README.md]: Also doesn't this PR make the bit in the docs about the selfplay target obsolete?

File: selfplay.py [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+                   'What percent of games to disable resign for.')
+
+# this should be called "verbosity" but flag name conflicts with absl.logging.
+flags.DEFINE_integer('verbose', 1, 'How much debug info to print.')
+
+FLAGS = flags.FLAGS
+
+def play(network, verbosity=0):
+    ''' Plays out a self-play match, returning a MCTSPlayer object wrapping:
+        - the final position
```
Comment (amj) [inline selfplay.py]: s/wrapping/containing?

File: README.md (hunk 1 above)
Comment (brilee) [reply README.md]: fixed

File: gtp.py (hunk 2 above)
Comment (brilee) [reply gtp.py]: i'll change history just for you :P

Comment (brilee) [reply gtp.py]: no but actually, I blame git diff

https://github.com/tensorflow/minigo/pull/341/commits/1a6f8fdd3dd1db6d15dcca9b4660297e8730c51e

File: gtp.py (hunk 3 above)
Comment (brilee) [reply gtp.py]: You mean like, "this is just how absl.app works?"

File: selfplay.py (hunk 4 above)
Comment (brilee) [reply selfplay.py]: done

File: gtp.py (hunk 3 above)
Comment (amj) [reply gtp.py]: lol ok.  i have avoided looking up absl.app so far my whole life but... (long suffering sigh)... ok. :)

**Comment by amj** [comment]
LGTM.

**Pull request merged**